src/main/java/io/supertokens/webserver/api/session/SessionUserAPI.java

```diff
@@ -70,13 +71,15 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
 
         try {
             String[] sessionHandles;
-            TenantIdentifier tenantIdentifier = getTenantIdentifier(req);
-            Storage storage = getTenantStorage(req);
 
             if (fetchAcrossAllTenants) {
+                AppIdentifier appIdentifier = getAppIdentifier(req);
```

Comment:
we change this to find the right storage for the user, without enforcing the public tenant

Reply:
Copy the code of enforcePublicTenantAndGetStorageAndUserIdMappingForAppSpecificApi, but do not enforce public tenant

Reply:
Done